test/src/kbasesearchengine/test/main/IndexerWorkerTest.java

```diff
@@ -1138,6 +1139,188 @@ public void skipEvent() throws Exception {
+                .resolve("IndexerWorkerTest");
+        deleteRecursively(tempDir);
+
+        when(ws.getStorageCode()).thenReturn("code");
+
+        final StorageObjectType storageObjectType = StorageObjectType
+                .fromNullableVersion("code", "KBaseGenome.Genome", 1);
+
+        final IndexerWorkerConfigurator.Builder wrkCfg = IndexerWorkerConfigurator.getBuilder(
+                "myid", tempDir, logger)
+                .withStorage(storage, typeStore, idxStore)
+                .withEventHandler(ws);
+
+        final IndexerWorker worker = new IndexerWorker(wrkCfg.build());
+
+        final ObjectTypeParsingRules rule = ObjectTypeParsingRules.getBuilder(
+                new SearchObjectType("Genome", 1), storageObjectType)
+                .withIndexingRule(IndexingRules.fromPath(new ObjectJsonPath("assy_ref"))
+                        .build())
+                .build();
+
+        when(typeStore.listObjectTypeParsingRules(storageObjectType)).thenReturn(set(rule));
+
+        final Map<String, Object> data = ImmutableMap.of(
+                "thingy", 1,
+                "thingy2", "foo",
+                "subobjs", Arrays.asList(
+                        ImmutableMap.of("id", "an id", "somedata", "data"),
+                        ImmutableMap.of("id", "an id2", "somedata", "data2")
+                )
+        );
+
+        when(ws.load(eq(Arrays.asList(guid)), any(Path.class)))
+                .thenAnswer(new Answer<SourceData>() {
+
+                    @Override
+                    public SourceData answer(final InvocationOnMock inv) throws Throwable {
+                        final Path path = inv.getArgument(1);
+                        new ObjectMapper().writeValue(path.toFile(), data);
+                        return SourceData.getBuilder(
+                                new UObject(path.toFile()), "myobj", "somedude")
+                                .withNullableMD5("md5")
+                                .build();
+                    }
+                });
+
+        // don't set optional field overwriteExistingData (should default to false)
+        ChildStatusEvent event = new ChildStatusEvent(StatusEvent.getBuilder(
+                storageObjectType, Instant.ofEpochMilli(10000),
+                StatusEventType.NEW_VERSION)
+                .withNullableAccessGroupID(1)
+                .withNullableObjectID("2")
+                .withNullableVersion(3)
+                .withNullableisPublic(false)
+                .build(),
+                new StatusEventID("pid"));
+
+        when(ws.updateObjectEvent(event.getEvent())).thenReturn(event.getEvent());
+
+        // simulate to show that index record for corresponding event already exists
+        when(idxStore.checkParentGuidsExist(set(guid))).thenReturn(ImmutableMap.of(guid, true));
+
+        final StatusEventProcessingState res = worker.processEvent(event);
+        assertThat("incorrect state", res, is(StatusEventProcessingState.INDX));
+
```

Comment:
Should check that there was no interactions with the indexing storage system other than checking the parent guid

Reply:
added.